cmake/external/leveldb.cmake

```diff
@@ -0,0 +1,59 @@
+# Copyright 2017 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+include(ExternalProject)
+
+# LevelDB does not build on Windows (yet)
+# See:
+#   https://github.com/google/leveldb/issues/363
+#   https://github.com/google/leveldb/issues/466
+if(NOT WIN32)
+  # Clean up warning output to reduce noise in the build
+  if(CMAKE_CXX_COMPILER_ID MATCHES "Clang|GNU")
+    set(
+      LEVELDB_CXX_FLAGS "\
+        -Wno-deprecated-declarations"
+    )
+  endif()
+
+  # Map CMake compiler configuration down onto the leveldb Makefile
+  set(
+    LEVELDB_OPT "\
+      $<$<CONFIG:Debug>:${CMAKE_CXX_FLAGS_DEBUG}> \
+      $<$<CONFIG:Release>:${CMAKE_CXX_FLAGS_RELEASE}>"
+  )
+
+  ExternalProject_Add(
+    leveldb
+
+    GIT_REPOSITORY "git@github.com:google/leveldb.git"
```

Comment:
The https url might be a better choice as people who anonymously checkout the code won't have to deal with ssh keys. (Assuming https checkouts are supported here, and also assuming we don't expect to make many/any changes to leveldb.)

https://github.com/google/leveldb.git

Reply:
Good idea. Done.